src/elements/position_offset.cpp

```diff
@@ -0,0 +1,169 @@
+      detail::ParameterTraits<ElevationOffset>::tag) const {
+    return elevation_ == boost::none;
+  }
+  bool SphericalPositionOffset::isDefault(
+      detail::ParameterTraits<DistanceOffset>::tag) const {
+    return distance_ == boost::none;
+  }
+
+  // --- Unsetter --- //
+  void SphericalPositionOffset::unset(
+      detail::ParameterTraits<AzimuthOffset>::tag) {
+    azimuth_ = boost::none;
+  }
+  void SphericalPositionOffset::unset(
+      detail::ParameterTraits<ElevationOffset>::tag) {
+    elevation_ = boost::none;
+  }
+  void SphericalPositionOffset::unset(
+      detail::ParameterTraits<DistanceOffset>::tag) {
+    distance_ = boost::none;
+  }
+
+  // ---- CARTESIAN POSITION OFFSET ---- //
+
+  // ---- Constructor ---- //
+
+  // ---- Defaults ---- //
+  const XOffset CartesianPositionOffset::xDefault_ = XOffset(0.f);
+  const YOffset CartesianPositionOffset::yDefault_ = YOffset(0.f);
+  const ZOffset CartesianPositionOffset::zDefault_ = ZOffset(0.f);
+
+  // ---- Getter ---- //
+  XOffset CartesianPositionOffset::get(
+      detail::ParameterTraits<XOffset>::tag) const {
+    return boost::get_optional_value_or(x_, xDefault_);
+  }
+  YOffset CartesianPositionOffset::get(
+      detail::ParameterTraits<YOffset>::tag) const {
+    return boost::get_optional_value_or(y_, yDefault_);
+  }
+  ZOffset CartesianPositionOffset::get(
+      detail::ParameterTraits<ZOffset>::tag) const {
+    return boost::get_optional_value_or(z_, zDefault_);
+  }
+
+  // ---- Has ---- //
+  bool CartesianPositionOffset::has(
+      detail::ParameterTraits<XOffset>::tag) const {
+    return true;
+  }
+  bool CartesianPositionOffset::has(
+      detail::ParameterTraits<YOffset>::tag) const {
+    return true;
+  }
+  bool CartesianPositionOffset::has(
+      detail::ParameterTraits<ZOffset>::tag) const {
+    return true;
+  }
+
+  // ---- Setter ---- //
+  void CartesianPositionOffset::set(XOffset x) { x_ = x; }
+  void CartesianPositionOffset::set(YOffset y) { y_ = y; }
+  void CartesianPositionOffset::set(ZOffset z) { z_ = z; }
+
+  // ---- isDefault ---- //
+  bool CartesianPositionOffset::isDefault(
+      detail::ParameterTraits<XOffset>::tag) const {
+    return x_ == boost::none;
+  }
+  bool CartesianPositionOffset::isDefault(
+      detail::ParameterTraits<YOffset>::tag) const {
+    return y_ == boost::none;
+  }
+  bool CartesianPositionOffset::isDefault(
+      detail::ParameterTraits<ZOffset>::tag) const {
+    return z_ == boost::none;
+  }
+
+  // --- Unsetter --- //
+  void CartesianPositionOffset::unset(detail::ParameterTraits<XOffset>::tag) {
+    x_ = boost::none;
+  }
+  void CartesianPositionOffset::unset(detail::ParameterTraits<YOffset>::tag) {
+    y_ = boost::none;
+  }
+  void CartesianPositionOffset::unset(detail::ParameterTraits<ZOffset>::tag) {
+    z_ = boost::none;
+  }
+
+  // ---- FREE FUNCTIONS ---- //
+
+  bool isSpherical(const PositionOffset& offset) {
+    if (offset.which() == 0) {
```

Comment:
I know this is the same as the in `Position`, but I'd rather `.which()` was not used to test variant types, as it depends on the order. `return offset.type() == typeid(SphericalPositionOffset);` would be my personal preference (same below).